src/IntegrationUtils.ts

```diff
@@ -59,13 +60,55 @@ const choiceList = [
 
  let childProcessMap : Map<string, child_process.ChildProcess>;
 
- export function startIntegration(context: vscode.Uri): Promise<boolean> {
+ export function startIntegration(...args: any[]): Promise<boolean> {
```

Comment:
I am wondering if we should not better do `startIntegration(context: vscode.Uri, ...args: any[])` instead to keep the didact in a separate flow.

Reply:
I can definitely try that.

Reply:
So, regardless of what I put in the function argument declaration, the first argument always comes through as an array of arguments. It appears that execute isn't distinguishing between a context-only argument list and a mix of arguments... it's always passed an array of any[]...

`executeCommand<T>(command: string, ...rest: any[]): Thenable<T | undefined>`

We're getting the "rest" passed to the command function triggered when the command Id is called.

Reply:
ah, my fault. Now I see that this is called directly by the command.